### ISSUE TITLE
allowDestructuring option for no-sparse-arrays rule

### ISSUE THREAD

**Issue by caub** [eslint/eslint#12463]
**What rule do you want to change?**
no-sparse-arrays

**Does this change cause the rule to produce more or fewer warnings?**
fewer warnings

**How will the change be implemented? (New option, new default behavior, etc.)?**
New option

**Please provide some example code that this change will affect:**

```js
const [, y] = [1, 2];
```

**What does the rule currently do for this code?**
`error    Unexpected comma in middle of array                    no-sparse-arrays`

**What will the rule do after it's changed?**
it'll allow it, with `"no-sparse-array": ["error", {"allowDestructuring": true}]`

**Are you willing to submit a pull request to implement this change?**
I can try, yes

**Comment by ljharb** [comment]
This shouldn’t require an option - it’s broken, since omitting items in array destructuring is in no way “sparse arrays”. This rule simply shouldn’t consider array spreads.

**Comment by jaawerth** [comment]
Yeah this just seems like a bug, since destructuring isn't creating an array, sparse or otherwise.

**Comment by mdjermanovic** [comment]
Hi @caub, thanks for the issue!

This would be a bug, but I can't reproduce it: [Online Demo Link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXNwYXJzZS1hcnJheXM6IFtcImVycm9yXCJdICovXG5cbmNvbnN0IFssIHldID0gWzEsIDJdO1xuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

It might be good to fill the bug report issue template.

**Comment by caub** [comment]
Oh, sorry guys, I had an actual sparse array further right in the error line

```js
const [, k, v] = str.match(/^(\w+) (\S+)/) || [, 'request', str)];
```
sorry for that false alert

**Comment by mdjermanovic** [comment]
Thanks! I'm closing the issue then.
